discord/commands/core.py

```diff
@@ -826,11 +828,14 @@ async def _invoke(self, ctx: ApplicationContext) -> None:
                             kw["guild"] = ctx.guild
                         elif op.input_type is SlashCommandOptionType.channel:
                             # NOTE:
-                            # This is a fallback in case the channel is not found in the guild's channels.
-                            # If this fallback occurs, at the very minimum, permissions will be incorrect
-                            # due to a lack of permission_overwrite data.
-                            obj_type = _guild_channel_factory(_data["type"])[0]
-                            kw["guild"] = ctx.guild
+                            # This is a fallback in case the channel/thread is not found in the
+                            # guild's channels/threads. For channels, if this fallback occurs, at the very minimum,
+                            # permissions will be incorrect due to a lack of permission_overwrite data.
+                            # For threads, if this fallback occurs, info like thread owner id, message count,
+                            # flags, and more will be missing due to a lack of data sent by Discord.
+                            obj_type = _threaded_guild_channel_factory(_data["type"])[0]
+                            if op._raw_type is not Thread:
```

Comment:
You shouldn't rely on `_raw_type` here, I would instead either check `obj_type` or just add the key anyways, as an extra key won't hurt.